src/PSR7/Validators/SerializedParameter.php

```diff
@@ -125,8 +125,13 @@ private function isJsonContentType(): bool
      */
     private function castToSchemaType($value, ?string $type)
     {
-        if (($type === CebeType::BOOLEAN) && is_scalar($value) && preg_match('#^(true|false)$#i', (string) $value)) {
-            return is_string($value) ? strtolower($value) === 'true' : (bool) $value;
+        if ($type === CebeType::BOOLEAN && is_scalar($value)) {
+            if (preg_match('#^(true|false)$#i', (string)$value)) {
+                return is_string($value) ? strtolower($value) === 'true' : (bool)$value;
```

Comment:
Looks like there is no need to return `(bool) $value here anymore, since value is either string `true` or `string` false here. I think this didn't work before either for the same reason